controllers/buildpipeline/buildpipeline_adapter.go

```diff
@@ -64,12 +65,12 @@ func NewAdapter(pipelineRun *tektonv1beta1.PipelineRun, component *applicationap
 // to the build PipelineRun being processed exists. Otherwise, it will create a new pipeline Snapshot.
 func (a *Adapter) EnsureSnapshotExists() (controller.OperationResult, error) {
 	if !h.HasPipelineRunSucceeded(a.pipelineRun) {
+		a.logger.Info("Build pipeline has not succeeded")
```

Comment:
Should we also add some code to remove finalizer from the failed build plr? Otherwise I guess the failed plr with a finalizer won't be deleted.

Reply:
You're right, I think we should call `removeFinalizerAndContinueProcessing()` here.  To prevent a case where we prematurely remove the finalizer, we'll need to call `HasPipelineRunFinished` either below line 66 or at the beginning of `removeFinalizerAndContinueProcessing()`.  Do you have an opinion as to which we should do?

@dheerajodha we had discussed something similar in another comment so feel free to weigh in here too.

Reply:
Calling `HasPipelineRunFinished` below line 66 looks good to me.

Reply:
maybe something like this?
```
if !h.HasPipelineRunSucceeded(a.pipelineRun) {
          if HasPipelineRunFinished(a.pipelineRun) {
                 // This means that a.pipelineRun has "Failed"
                return removeFinalizerAndContinueProcessing()
          }
          a.logger.Info("Build pipeline has not succeeded YET")
          return controller.ContinueProcessing()
}
```